### PR TITLE
address latest clippy warnings

### PR DESCRIPTION
**Pull request by bmc-msft**

### CODE REVIEW

File: src/agent/onefuzz-supervisor/src/buffer.rs [hunk 1]
```diff
@@ -54,22 +54,23 @@ mod tests {
 
         assert!(buf.data().is_empty());
 
-        buf.write(&[1, 2, 3]).unwrap();
+        buf.write_all(&[1, 2, 3]).unwrap();
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/buffer.rs:57]: Our impl of `write()` is actually the method under test, so I think it is most appropriate to suppress the linter here. `Write::write_all()` ends up being equivalent, but it _would_ mask a change.

File: src/agent/onefuzz/src/libfuzzer.rs [hunk 2]
```diff
@@ -280,6 +280,10 @@ mod tests {
             .expect("no captures");
 
         assert_eq!(parsed.iters(), 2097152);
-        assert_eq!(parsed.execs_sec(), 699050.0_f64);
+
+        let error_margin = std::f64::EPSILON;
+        let expected: f64 = 699050.0;
+        let execs_sec = parsed.execs_sec();
```
Comment (ranweiler) [inline src/agent/onefuzz/src/libfuzzer.rs:286]:
```suggestion
        let execs_sec = parsed.execs_sec();
        assert!(execs_sec.is_finite());
```

File: src/agent/onefuzz/src/libfuzzer.rs [hunk 3]
```diff
@@ -280,6 +280,10 @@ mod tests {
...
+        assert!((execs_sec - expected).abs() < error_margin);
```
Comment (ranweiler) [inline src/agent/onefuzz/src/libfuzzer.rs:287]: I'd suggest just using the epsilon value directly:
```suggestion
        assert!((execs_sec - expected).abs() < f64::EPSILON);
```
I'll note that clippy suggests using this pattern for approximate equality, but the doc it links to advises AGAINST doing it. 😑

**Review by ranweiler — APPROVED** [review]

File: src/agent/onefuzz-supervisor/src/buffer.rs (hunk 1 above)
Comment (bmc-msft) [reply src/agent/onefuzz-supervisor/src/buffer.rs:57]: If our implementation of write is under test, then should we test the result?

Comment (ranweiler) [reply src/agent/onefuzz-supervisor/src/buffer.rs:57]: I think so, so we can be sure it plays nice with wrappers. So for example, I'd suggest that this line becomes:
```suggestion
        assert_eq!(buf.write(&[1, 2, 3]), Ok(3));
```

File: src/agent/onefuzz-supervisor/src/buffer.rs [hunk 4]
```diff
@@ -54,22 +54,23 @@ mod tests {
...
         assert_eq!(buf.data(), &[1, 2, 3]);
 
-        buf.write(&[]).unwrap();
+        buf.write_all(&[]).unwrap();
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/buffer.rs:60]:
```suggestion
        assert_eq!(buf.write(&[]), Ok(0));
```

File: src/agent/onefuzz-supervisor/src/buffer.rs [hunk 5]
```diff
@@ -54,22 +54,23 @@ mod tests {
...
         assert_eq!(buf.data(), &[1, 2, 3]);
 
-        buf.write(&[4, 5]).unwrap();
+        buf.write_all(&[4, 5]).unwrap();
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/buffer.rs:63]:
```suggestion
        assert_eq!(buf.write(&[4, 5]), Ok(2));        
```

File: src/agent/onefuzz-supervisor/src/buffer.rs [hunk 6]
```diff
@@ -54,22 +54,23 @@ mod tests {
...
         assert_eq!(buf.data(), &[1, 2, 3, 4, 5]);
 
-        buf.write(&[6, 7, 8]).unwrap();
+        buf.write_all(&[6, 7, 8]).unwrap();
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/buffer.rs:66]:
```suggestion
        assert_eq!(buf.write(&[6, 7, 8]), Ok(3));
```

File: src/agent/onefuzz-supervisor/src/buffer.rs [hunk 7]
```diff
@@ -54,22 +54,23 @@ mod tests {
...
         assert_eq!(buf.data(), &[4, 5, 6, 7, 8]);
 
-        buf.write(&[9, 10, 11, 12, 13]).unwrap();
+        buf.write_all(&[9, 10, 11, 12, 13]).unwrap();
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/buffer.rs:69]:
```suggestion
        assert_eq!(buf.write(&[9, 10, 11, 12, 13]), Ok(5));
```

File: src/agent/onefuzz-supervisor/src/buffer.rs [hunk 8]
```diff
@@ -54,22 +54,23 @@ mod tests {
...
         assert_eq!(buf.data(), &[9, 10, 11, 12, 13]);
 
-        buf.write(&[14, 15, 16, 17, 18, 19, 20, 21, 22]).unwrap();
+        buf.write_all(&[14, 15, 16, 17, 18, 19, 20, 21, 22])
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/buffer.rs:72]:
```suggestion
        assert_eq!(buf.write(&[14, 15, 16, 17, 18, 19, 20, 21, 22]), Ok(9));
```

File: src/agent/onefuzz-supervisor/src/buffer.rs [hunk 9]
```diff
@@ -54,22 +54,23 @@ mod tests {
...
+        assert_eq!(buf.write(&[6, 7, 8]), Ok(3));
         assert_eq!(buf.data(), &[4, 5, 6, 7, 8]);
 
-        buf.write(&[9, 10, 11, 12, 13]).unwrap();
+        assert_eq!(buf.write(&[9, 10, 11, 12, 13]), Ok(5));
         assert_eq!(buf.data(), &[9, 10, 11, 12, 13]);
 
-        buf.write(&[14, 15, 16, 17, 18, 19, 20, 21, 22]).unwrap();
+        assert_eq!(buf.write(&[14, 15, 16, 17, 18, 19, 20, 21, 22]), Ok(9));
+            .unwrap();
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/buffer.rs:73]:
```suggestion
```

File: src/agent/onefuzz/src/libfuzzer.rs [hunk 10]
```diff
@@ -280,6 +280,11 @@ mod tests {
             .expect("no captures");
 
         assert_eq!(parsed.iters(), 2097152);
-        assert_eq!(parsed.execs_sec(), 699050.0_f64);
+
+        let error_margin = std::f64::EPSILON;
```
Comment (bmc-msft) [inline src/agent/onefuzz/src/libfuzzer.rs:284]:
```suggestion
```

**Pull request merged**